mr/src/test/java/org/apache/iceberg/mr/hive/TestTables.java

```diff
@@ -189,16 +189,37 @@ public Table createTable(TestHiveShell shell, String tableName, Schema schema, F
    */
   public Table createTable(TestHiveShell shell, String tableName, Schema schema, PartitionSpec spec,
       FileFormat fileFormat, List<Record> records)  {
+    return createTable(shell, tableName, schema, spec, fileFormat, records, null);
+  }
+
+  /**
+   * Creates a partitioned Hive test table using Hive SQL. The table will be in the 'default' database.
+   * The table will be populated with the provided List of {@link Record}s using a Hive insert statement.
+   * @param shell The HiveShell used for Hive table creation
+   * @param tableName The name of the test table
+   * @param schema The schema used for the table creation
+   * @param spec The partition specification for the table
+   * @param fileFormat The file format used for writing the data
+   * @param records The records with which the table is populated
+   * @param properties Additional TblProperties to pass to the create table statement
+   * @return The created table
+   * @throws IOException If there is an error writing data
+   */
+  public Table createTable(TestHiveShell shell, String tableName, Schema schema, PartitionSpec spec,
```

Comment:
Could we base this change the already existing hive code?
https://github.com/apache/hive/blob/90428cc5f594bd0abb457e4e5c391007b2ad1cb8/iceberg/iceberg-handler/src/test/java/org/apache/iceberg/mr/hive/TestTables.java#L338-L356

Reply:
You mean I should put this test framework change into the Hive repo as well?
I can do so, I think the timing would be better after this PR is merged. Unless if there is any restriction that changes like this should land in Hive first.